From 83ab73384fc1d589b7b99278241c97992ecb15ac Mon Sep 17 00:00:00 2001
From: Alex Marginean <alexandru.marginean@nxp.com>
Date: Thu, 30 Aug 2018 13:59:46 +0300
Subject: [PATCH 284/706] enetc: fix a couple of endianness issues in CBDR

Signed-off-by: Alex Marginean <alexandru.marginean@nxp.com>
(cherry picked from commit 203f16300025741acca957fadd8b3de71452c188)
(cherry picked from commit c3ec54a560f98d17478c181ba84d060359bdfd2c)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 drivers/net/ethernet/freescale/enetc/enetc_cbdr.c | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/drivers/net/ethernet/freescale/enetc/enetc_cbdr.c b/drivers/net/ethernet/freescale/enetc/enetc_cbdr.c
index a2566df3c373..8f08706b8c51 100644
--- a/drivers/net/ethernet/freescale/enetc/enetc_cbdr.c
+++ b/drivers/net/ethernet/freescale/enetc/enetc_cbdr.c
@@ -190,8 +190,8 @@ int enetc_set_fs_entry(struct enetc_si *si, struct enetc_cmd_rfse *rfse,
 	tmp_align = PTR_ALIGN(tmp, RFSE_ALIGN);
 	memcpy(tmp_align, rfse, sizeof(*rfse));
 
-	cbd.addr[0] = lower_32_bits(dma_align);
-	cbd.addr[1] = upper_32_bits(dma_align);
+	cbd.addr[0] = cpu_to_le32(lower_32_bits(dma_align));
+	cbd.addr[1] = cpu_to_le32(upper_32_bits(dma_align));
 
 	if (async)
 		cbd.status_flags |= ENETC_CBD_FLAGS_IE;
@@ -235,10 +235,10 @@ static int enetc_cmd_rss_table(struct enetc_si *si, u32 *table, int count,
 	/* fill up the descriptor */
 	cbd.cmd = read ? 2 : 1;
 	cbd.cls = 3;
-	cbd.length = count;
+	cbd.length = cpu_to_le16(count);
 
-	cbd.addr[0] = lower_32_bits(dma_align);
-	cbd.addr[1] = upper_32_bits(dma_align);
+	cbd.addr[0] = cpu_to_le32(lower_32_bits(dma_align));
+	cbd.addr[1] = cpu_to_le32(upper_32_bits(dma_align));
 
 	err = enetc_send_cmd(si, &cbd, false);
 	if (err)
-- 
2.17.1

